.github/scripts/changelog_checker

```diff
@@ -0,0 +1,58 @@
+#!/bin/sh
+
+set -ue
+
+UNTRACKED="
+appveyor_build.cmd
+appveyor.patch
+appveyor_test.sh
+appveyor.yml
+AUTHORS
+CONTRIBUTING.md
+.gitattributes
+.github
+.gitignore
+ISSUE_TEMPLATE.md
+.ocamlinit
+.ocp-indent
+.ocplint
+README.md
+.travis-ci.sh
+.travis.yml
+master_changes.md
+"
+changelog=master_changes.md
+diffile=/tmp/diff
+
+git fetch origin $GITHUB_BASE_REF --depth=1 --quiet
+echo "> base commit"
+git show origin/$GITHUB_BASE_REF --format=oneline -s
+
+git diff origin/$GITHUB_BASE_REF --name-status --diff-filter=AM > $diffile
+updated=0
+if [ -z "`grep -s $changelog $diffile`" ] ; then
+  updated=1
+fi
+
+echo "> all changes"
+cat $diffile
+
+for untrack in $UNTRACKED ; do
+  sed -i "/$untrack/d" $diffile
```

Comment:
This is risky - all those patterns are treated as though *$untrack* - shouldn't the patterns at least match at the start only?

Reply:
("risky" in terms of the utter evil of escaping the change logging checking action!)